source/guide_salt.rst

```diff
@@ -0,0 +1,134 @@
+.. author:: ctr49 <https://github.com/ctr49>
+
+.. tag:: automation
+
+.. highlight:: console
+
+.. sidebar:: Logo
+
+  .. image:: _static/images/saltstack.png
+      :align: center
+
+##########
+SaltStack
+##########
+
+.. tag_list::
+
+`SaltStack`_ (or simply `Salt`) is Python-based, open-source software for event-driven IT automation, remote task execution, and configuration management. Supporting the "Infrastructure as Code" approach to data center system and network deployment and management, configuration automation, SecOps orchestration, vulnerability remediation, and hybrid cloud control.
+
+The `salt-master` controls so called `salt-minions` through formulae, pillars and grains. The scope of this guide is to install a `salt-master`(without a co-located minion) on an Uberspace.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * :manual:`Opening ports <basics-ports>`
+  * :manual:`supervisord <daemons-supervisord>`
+
+Prerequisites
+=============
+
+We need two open TCP ports for minions to communicate with the Salt master.
+
+::
+
+ [user@host ~]$ uberspace port add
+ Port <high-port1> will be open for TCP and UDP traffic in a few minutes.
+ [user@host ~]$ uberspace port add
+ Port <high-port2> will be open for TCP and UDP traffic in a few minutes.
+ [user@host ~]$ uberspace port list
+ <high-port1>
+ <high-port2>
+
+Take note of the high ports, you will need them for salt master configuration later.
+
+Installation
+============
+
+Install salt-master
+-------------------
+
+The `salt` installation script requires root privileges and only supports virtualenv installation for Ubuntu. So we'll follow the `Developer installation` guide to install the salt master manually. However, we don't want the latest head development version, but the latest stable/tagged version. As of writing this guide, the latest stable version is 2019.2.1
+
+::
+
+ [user@host ~]$ mkdir -p ~/salt/src
+ [user@host ~]$ cd ~/salt/src
+ [user@host ~]$ git clone https://github.com/saltstack/salt
+ [user@host ~]$ cd salt
+ [user@host ~]$ git remote add upstream https://github.com/saltstack/salt
+ [user@host ~]$ git fetch --tags upstream
+ [user@host ~]$ git branch 2019.2.1
```

Comment:
```suggestion
 [user@host ~]$ git checkout 2019.2.1
```

[`git branch`](https://git-scm.com/docs/git-branch) creates a branch instead of switching to an existing one. Did you mean [`git checkout`](https://git-scm.com/docs/git-checkout)?

Reply:
right this was writing things from memory, should be "git checkout tags/v2019.2.0", already corrected